src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
Yep – I should have written in the description I did in fact take some shortcuts – but I agree, this is tech-debt we can and should address now.